storage/inkless/src/test/java/io/aiven/inkless/consume/FetchPlannerTest.java

```diff
@@ -210,19 +225,391 @@ private FetchPlanner fetchPlannerJob(Map<TopicIdPartition, FindBatchResponse> ba
+            // For this test, we only care about the final data returned by readToByteBuffer()
+            when(fetcher.fetch(eq(OBJECT_KEY_A), any(ByteRange.class)))
+                .thenReturn(null); // Return value doesn't matter, readToByteBuffer() is also mocked
+            when(fetcher.fetch(eq(OBJECT_KEY_B), any(ByteRange.class)))
+                .thenReturn(null); // Return value doesn't matter, readToByteBuffer() is also mocked
+
+            // Mock readToByteBuffer to return the test data we want to verify
+            // Order matters: first call returns dataA, second call returns dataB
+            when(fetcher.readToByteBuffer(any()))
+                .thenReturn(ByteBuffer.wrap(dataA))
+                .thenReturn(ByteBuffer.wrap(dataB));
+
+            final Map<TopicIdPartition, FindBatchResponse> coordinates = Map.of(
+                partition0, FindBatchResponse.success(List.of(
+                    new BatchInfo(1L, OBJECT_KEY_A.value(),
+                        BatchMetadata.of(partition0, 0, 10, 0, 0, 10, 20, TimestampType.CREATE_TIME)),
+                    new BatchInfo(2L, OBJECT_KEY_B.value(),
+                        BatchMetadata.of(partition0, 0, 10, 1, 1, 11, 21, TimestampType.CREATE_TIME))
+                ), 0, 2)
+            );
+
+            final FetchPlanner planner = new FetchPlanner(
+                time, OBJECT_KEY_CREATOR, keyAlignmentStrategy,
+                caffeineCache, fetcher, dataExecutor, coordinates, metrics
+            );
+
+            // Execute: Trigger fetch operations
+            final List<CompletableFuture<FileExtent>> futures = planner.get();
+
+            // Verify: Should have two futures
+            assertThat(futures).hasSize(2);
+
+            // Wait for all to complete
+            CompletableFuture.allOf(futures.toArray(new CompletableFuture[0])).get();
+
+            // Verify both were fetched
+            verify(fetcher).fetch(eq(OBJECT_KEY_A), any(ByteRange.class));
+            verify(fetcher).fetch(eq(OBJECT_KEY_B), any(ByteRange.class));
+
+            // Verify correct data for each
+            final List<FileExtent> results = futures.stream()
+                .map(f -> {
+                    try {
+                        return f.get();
+                    } catch (Exception e) {
+                        throw new RuntimeException(e);
+                    }
+                })
+                .collect(Collectors.toList());
+
+            assertThat(results).hasSize(2);
+
+            // Verify the actual data content matches expected values
+            // Note: We cannot rely on ordering since the futures complete asynchronously,
+            // so we use containsExactlyInAnyOrder to verify both byte arrays are present.
+            assertThat(results)
+                .extracting(FileExtent::data)
+                .containsExactlyInAnyOrder(dataA, dataB);
```

Comment:
Could this be just a one-liner plus verification?
```suggestion
            assertThat(planner.get()).map(CompletableFuture::get).map(FileExtent::data)
                .containsExactlyInAnyOrder(dataA, dataB);
            verify(fetcher).fetch(eq(OBJECT_KEY_A), any(ByteRange.class));
            verify(fetcher).fetch(eq(OBJECT_KEY_B), any(ByteRange.class));
```